File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/user/UserStatus.java
package br.com.deltaglobalbank.identity.domain.user;

import java.util.Arrays;

public enum UserStatus {
    ACTIVE,
    SUSPENDED,
    LOCKED;

    public String toDatabaseValue() {
        return name().toLowerCase();
    }

    public static UserStatus fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(status -> status.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("Status de user inválido: " + value));
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/user/UserStatus.kt
package br.com.deltaglobalbank.identity.domain.user

enum class UserStatus {
    ACTIVE,
    SUSPENDED,
    LOCKED;

    fun toDatabaseValue(): String = name.lowercase()

    companion object {
        fun fromDatabaseValue(value: String): UserStatus =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("Status de user inválido: $value")
    }
}